### DISCUSSION TITLE
FSharpFunc + ToString() Analyzer suggestion

### DISCUSSION THREAD

**Discussion by En3Tho** [dotnet/fsharp#11166] [Ideas]
While I was skimming through our code I found a funny bug: someone wrote Guid.NewGuid.ToString() and forgot to put those "()" after NewGuid call. Can we have an analyzer to find such calls (at least string/ToString) on FSharpFuncs?

**Comment by cartermp** [comment]
Interesting, yeah that could be a good analyzer. More generally, this would probably be:

_A method or F# function that accepts `unit` as input that does not have a `unit` as the parameter_ emits a warning at the call site. Then a quick fix would just be to add `()` at the end.
